src/components/MiniDrawer.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  type CSSObject,
+  Divider,
+  Drawer,
+  IconButton,
+  type Theme,
+} from "@mui/material"
+import { ChevronLeft, Menu } from "@mui/icons-material"
+import { type FC, type ReactNode } from "react"
+
+const DRAWER_WIDTH = 240
+
+const openedMixin = (theme: Theme): CSSObject => ({
+  width: DRAWER_WIDTH,
+  transition: theme.transitions.create("width", {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.enteringScreen,
+  }),
+  overflowX: "hidden",
+})
+
+const closedMixin = (theme: Theme): CSSObject => ({
+  transition: theme.transitions.create("width", {
+    easing: theme.transitions.easing.sharp,
+    duration: theme.transitions.duration.leavingScreen,
+  }),
+  overflowX: "hidden",
+  width: `calc(${theme.spacing(7)} + 1px)`,
+  [theme.breakpoints.up("sm")]: {
+    width: `calc(${theme.spacing(8)} + 1px)`,
+  },
+})
+
+export interface MiniDrawerProps {
+  open: boolean
+  children: ReactNode
+  onToggle: () => void
+}
+
+const MiniDrawer: FC<MiniDrawerProps> = props => {
+  const { open, onToggle, children } = props
+  return (
+    <Drawer
+      variant="permanent"
+      open={open}
+      sx={theme => {
+        const base: CSSObject = {
+          width: DRAWER_WIDTH,
+          flexShrink: 0,
+          whiteSpace: "nowrap",
+          boxSizing: "border-box",
+          overflowX: "hidden",
+        }
+        if (open) {
+          return {
+            ...base,
+            ...openedMixin(theme),
+            "& .MuiDrawer-paper": openedMixin(theme),
+          }
+        } else {
+          return {
+            ...base,
+            ...closedMixin(theme),
+            "& .MuiDrawer-paper": closedMixin(theme),
+          }
+        }
+      }}
+    >
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "flex-end",
+          padding: "0 10px",
+        }}
+      >
```

Comment:
```suggestion
      <div
        style={{
          display: "flex",
          justifyContent: "flex-end",
          padding: "0 10px",
        }}
      >
```

I understand what your doing here - pushing to the right when the drawer is open - however, a few points:
1. don't use magic numbers (10px). [we have a global spacing defined](https://github.com/ocadotechnology/codeforlife-package-javascript/blob/main/src/theme/spacing.ts), which you should use instead. We have defined 1space=8px, so 10px=1.25space.
2. Instead of using `<div />` components you should use a `<Box/>` component from MUI instead. This is basically a div but powered by MUI's theme (it has the `sx` prop).